java/arcs/android/storage/database/AndroidSqliteDatabaseManager.kt

```diff
@@ -36,39 +30,14 @@ import kotlinx.coroutines.sync.withLock
  */
 class AndroidSqliteDatabaseManager(
     context: Context,
-    lifecycleParam: Lifecycle? = null,
     // Maximum size of the database file, if it surpasses this size, the database gets reset.
     private val maxDbSizeBytes: Int = MAX_DB_SIZE_BYTES
-) : DatabaseManager, LifecycleObserver {
+) : DatabaseManager {
     private val context = context.applicationContext
-    private val lifecycle = lifecycleParam ?: getLifecycle()
     private val mutex = Mutex()
     private val dbCache by guardedBy(mutex, mutableMapOf<DatabaseIdentifier, DatabaseImpl>())
     override val registry = AndroidSqliteDatabaseRegistry(context)
 
-    init {
-        lifecycle?.addObserver(this) ?: Log.debug {
-            "No lifecycle available for AndroidSqliteDatabaseManager with context $context"
-        }
-    }
-
-    /*
-     * Temporary hack workaround to avoid breaking G3 with a refactor. Followup will add
-     * explicit lifecycle parameter to ctor.
-     */
-    private fun getLifecycle(): Lifecycle? {
-        var lifecycleOwner = context
-        while (lifecycleOwner != null && lifecycleOwner !is LifecycleOwner &&
-            lifecycleOwner is ContextWrapper) {
-            lifecycleOwner = lifecycleOwner.baseContext
-        }
-
-        return (lifecycleOwner as? LifecycleOwner)?.lifecycle
-    }
-
-    @OnLifecycleEvent(Lifecycle.Event.ON_DESTROY)
-    fun onLifecycleDestroyed() = close()
-
```

Comment:
It seems the intent was to associate the database manager with the onDestroy of an application. Do we need that?

Reply:
The hope was that the manager might be bound to the lifecycle of the StorageService, rather than the application.. but I think our architecture has evolved such that that's no longer appropriate. This seems fine to remove.